anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -748,29 +754,60 @@ def record_count(self, group_name: str, db: Session) -> int:
         """
         return len(self._find_match(db))
 
-    def _flush_group(self, group_obj: FeedGroupMetadata) -> None:
+    def flush_group(self, group_name: str) -> FeedGroupMetadata:
         """
         Flush a specific data group. Do a db flush, but not a commit at the end to keep the transaction open.
 
-        :param group_obj: feed group metadata record
-        :type group_obj: FeedGroupMetadata
+        :param group_name: name of group to flush
+        :type group_name: str
+        :return: db record containing metadata of group flushed
+        :rtype: FeedGroupMetadata
         """
-        db = get_session()
-        catalog_client = self._catalog_client
+        raise NotImplementedError
 
-        logger.info(
-            self._log_context.format_msg(
-                "Flushing group records",
-            )
-        )
+    def group_by_name(self, group_name: str) -> Optional[FeedGroupMetadata]:
+        """
+        Gets the metadata record for the group with the given name.
 
-        records = self._find_match(db)
-        for record in records:
-            catalog_client.delete_document(self.__feed_name__, record.archive_checksum)
-            db.delete(record)
-        group_obj.last_sync = None  # Null the update timestamp to reflect the flush
-        group_obj.count = 0
-        db.commit()
+        :param group_name: the name of the group for which to retreive the metadata record
+        :type group_name: str
+        :return: the metadata record if it exists
+        :rtype: Optional[FeedGroupMetadata]
+        """
+        return None
+
+    def update_counts(self) -> None:
+        """
+        Self-contained unit of work to update the row counts for each feed group in the feed group metadata
+        """
+        return
+
+    def flush_all(self) -> FeedMetadata:
+        """
+        Flush all groups for the feed, unset the last full sync timestamp, but leave metadata records for feed.
+
+        :return: db record containing feed metadata
+        :rtype: FeedMetadata
+        """
+        db = get_session()
+        try:
+            catalog_client = self._catalog_client
+            records = self._find_match(db)
+            for record in records:
+                catalog_client.delete_document(
```

Comment:
Not 100% sure I understand the verbiage, but if what you are asking is whether or not we have a way of rolling back the catalog deletion, the answer is no. I can change this to cache the deleted records so that they may be added back on failure.

Reply:
```
    def flush_all(self) -> FeedMetadata:
        """
        Flush all groups for the feed, unset the last full sync timestamp, but leave metadata records for feed.

        :return: db record containing feed metadata
        :rtype: FeedMetadata
        """
        db = get_session()
        deleted_records = []
        try:
            records = self._get_db_metadata_records(db)
            for record in records:
                deleted_records.append(record.archive_checksum)
                db.delete(record)
            db.refresh(self.metadata)

            # Remove all groups
            self.metadata.groups = []
            self.metadata.last_full_sync = None
            db.commit()
        except Exception:
            db.rollback()
            raise
        catalog_client = self._catalog_client
        for archive_checksum in deleted_records:
            catalog_client.delete_document(self.__feed_name__, archive_checksum)
        return self.metadata
```

Reply:
is this better?

Reply:
committed that change